File: Linked Lists/find_loop.py
class Node:
    def __init__(self, val):
        self.val = val
        self.next = None

def find_loop(head):
    first = head.next
    second = head.next.next
    while first != second:
        first = first.next
        second = second.next.next
    first = head
    while first != second:
        first = first.next
        second = second.next
    return first

File: Linked Lists/findLoop.js
class LinkedList {
    constructor(value) {
      this.value = value;
      this.next = null;
    }
}
  
function findLoop(head) {
    let first = head.next
    let second = head.next.next
    while (first !== second) {
        first = first.next
        second = second.next.next
        
    }
    first = head
    while (first !== second) {
        first = first.next
        second = second.next
    }
    
    return first
}
  